src/main/java/com/ksm/wstbackoffice/service/CountryService.java

```diff
@@ -23,4 +31,29 @@ public CountryService(CountryRepository countryRepository, CountryMapper country
     public CountryDto findById(String id) {
         return countryMapper.toDto(countryRepository.findById(id).orElse(null));
     }
+
+    public Map<String, Map<String, ?>> findAllBy(Collection<String> nameStartsWithFilters) {
+        Map<String, Map<String, ?>> countriesByNameStartingWith = new HashMap<String, Map<String, ?>>();
+        Map<String, Collection<String>> detailsCountriesByNameStartingWith = new HashMap<String, Collection<String>>();
+        Map<String, Integer> totalCountriesByNameStartingWith = new HashMap<String, Integer>();
+
+        List<CountryDto> countryDtoList = findAll();
+
+        nameStartsWithFilters.stream()
+                .filter(p -> p.trim().length() > 0)
```

Comment:
I think you want to check here if your filter is not blank.
You can do it with isBlank method
https://howtodoinjava.com/java11/check-blank-string/#:~:text=isBlank()%20method%20to%20determine,isEmpty()%20method.